bigtable/tests/unit/test_batcher.py

```diff
@@ -0,0 +1,182 @@
+        rows[1].set_cell('cf1', b'c1', 2)
+        rows[2].set_cell('cf1', b'c1', 3)
+        rows[3].set_cell('cf1', b'c1', 4)
+
+        return rows
+
+    def test_constructor(self):
+        from google.cloud.bigtable.batcher import MutationsBatcher
+
+        credentials = _make_credentials()
+        client = self._make_client(project='project-id',
+                                   credentials=credentials, admin=True)
+
+        instance = client.instance(instance_id='instance-id')
+        table = self._make_table(self.TABLE_ID, instance)
+
+        mutation_batcher = MutationsBatcher(table)
+        self.assertEqual(table, mutation_batcher.table)
+
+    def test_add_row(self):
+        table = _Table(self.TABLE_NAME)
+        mutation_batcher = self._make_batcher(table)
+
+        rows = self._make_rows(table)
+
+        for row in rows:
+            mutation_batcher.mutate(row)
+
+        mutation_batcher.flush()
+
+        self.assertEqual(table.mutation_calls, 1)
+
+    def test_flush_with_no_rows(self):
+        table = _Table(self.TABLE_NAME)
+        mutation_batcher = self._make_batcher(table)
+        mutation_batcher.flush()
+
+        self.assertEqual(table.mutation_calls, 0)
+
+    def test_add_row_with_max_flush_count(self):
+        table = _Table(self.TABLE_NAME)
+        mutation_batcher = self._make_batcher(table, flush_count=3)
+
+        rows = self._make_rows(table)
+
+        for row in rows:
+            mutation_batcher.mutate(row)
+
+        mutation_batcher.flush()
+
+        self.assertEqual(table.mutation_calls, 2)
+
+    @mock.patch('google.cloud.bigtable.batcher.MAX_MUTATIONS', new=3)
+    def test_add_row_with_max_mutations_failure(self):
+        from google.cloud.bigtable.batcher import MaxMutaionsError
+
+        table = _Table(self.TABLE_NAME)
+        mutation_batcher = self._make_batcher(table)
+
+        rows = self._make_rows(table)
+        rows[0].set_cell('cf1', b'c1', 2)
+        rows[0].set_cell('cf1', b'c1', 3)
+        rows[0].set_cell('cf1', b'c1', 4)
+
+        with self.assertRaises(MaxMutaionsError):
+            mutation_batcher.mutate(rows[0])
+
+    @mock.patch('google.cloud.bigtable.batcher.MAX_MUTATIONS', new=3)
+    def test_add_row_with_max_mutations(self):
+        table = _Table(self.TABLE_NAME)
+        mutation_batcher = self._make_batcher(table)
+
+        rows = self._make_rows(table)
+
+        for row in rows:
+            mutation_batcher.mutate(row)
+
+        mutation_batcher.flush()
+
+        self.assertEqual(table.mutation_calls, 2)
+
+    def test_add_row_with_max_row_bytes(self):
+        table = _Table(self.TABLE_NAME)
+        mutation_batcher = self._make_batcher(table,
+                                              max_row_bytes=1 * 1024 * 1024)
+
+        number_of_bytes = 2 * 1024 * 1024
+        max_value = b'1' * number_of_bytes
+
+        rows = self._make_rows(table)
+        rows[0].set_cell('cf1', b'c1', max_value)
```

Comment:
Please add multiple slightly smaller rows rather than a single large one to test the logic.